include/slang/parsing/Preprocessor.h

```diff
@@ -57,6 +58,16 @@ struct SLANG_EXPORT PreprocessorOptions {
 
     /// A set of preprocessor directives to be ignored.
     flat_hash_set<std::string_view> ignoreDirectives;
+
+    /// A flag to enable the interpretation of non-standard line comment
+    /// pragmas within the preprocessor.
+    bool enableCompatPragmas = false;
```

Comment:
Is it assumed that these options can only be used from the API?

I think it would be useful to use them from the slang cli interface as well

Reply:
Right now I am setting these with a downstream patch to `Driver.cpp`. For yosys-slang I think we want it on by default, possibly with a cli option to change the regexes or disable it.

If we can agree on standard slang options for this (with yosys-slang having it on by default, vanilla slang having it off by default), that would be great